healthx/handler_test.go

```diff
@@ -36,49 +36,133 @@ import (
+				"test": func(r *http.Request) error {
+					return alive
+				},
 			},
-		},
-	}
-	router := httprouter.New()
-	handler.SetHealthRoutes(router, true)
-	handler.SetVersionRoutes(router)
-	ts := httptest.NewServer(router)
-	c := http.DefaultClient
-
-	var healthBody swaggerHealthStatus
-	response, err := c.Get(ts.URL + AliveCheckPath)
-	require.NoError(t, err)
-	require.EqualValues(t, http.StatusOK, response.StatusCode)
-	require.NoError(t, json.NewDecoder(response.Body).Decode(&healthBody))
-	assert.EqualValues(t, "ok", healthBody.Status)
-
-	var versionBody swaggerVersion
-	response, err = c.Get(ts.URL + VersionPath)
-	require.NoError(t, err)
-	require.EqualValues(t, http.StatusOK, response.StatusCode)
-	require.NoError(t, json.NewDecoder(response.Body).Decode(&versionBody))
-	require.EqualValues(t, versionBody.Version, handler.VersionString)
-
-	response, err = c.Get(ts.URL + ReadyCheckPath)
-	require.NoError(t, err)
-	require.EqualValues(t, http.StatusServiceUnavailable, response.StatusCode)
-	out, err := ioutil.ReadAll(response.Body)
-	require.NoError(t, err)
-	assert.EqualValues(t, "ok", healthBody.Status)
-	assert.Equal(t, `{"errors":{"test":"not alive"}}`, strings.TrimSpace(string(out)))
-
-	alive = nil
-	response, err = c.Get(ts.URL + ReadyCheckPath)
-	require.NoError(t, err)
-	require.EqualValues(t, http.StatusOK, response.StatusCode)
-	require.NoError(t, json.NewDecoder(response.Body).Decode(&versionBody))
-	require.EqualValues(t, versionBody.Version, handler.VersionString)
+		}
+
+		router := httprouter.New()
+
+		handler.SetHealthRoutes(router, true)
+		handler.SetVersionRoutes(router)
+
+		ts := httptest.NewServer(router)
+
+		c := http.DefaultClient
+
+		var healthBody swaggerHealthStatus
+		response, err := c.Get(ts.URL + AliveCheckPath)
+		require.NoError(t, err)
+		require.EqualValues(t, http.StatusOK, response.StatusCode)
+		require.NoError(t, json.NewDecoder(response.Body).Decode(&healthBody))
+		assert.EqualValues(t, "ok", healthBody.Status)
+
+		var versionBody swaggerVersion
+		response, err = c.Get(ts.URL + VersionPath)
+		require.NoError(t, err)
+		require.EqualValues(t, http.StatusOK, response.StatusCode)
+		require.NoError(t, json.NewDecoder(response.Body).Decode(&versionBody))
+		require.EqualValues(t, versionBody.Version, handler.VersionString)
+
+		response, err = c.Get(ts.URL + ReadyCheckPath)
+		require.NoError(t, err)
+		require.EqualValues(t, http.StatusServiceUnavailable, response.StatusCode)
+		out, err := ioutil.ReadAll(response.Body)
+		require.NoError(t, err)
+		assert.EqualValues(t, "ok", healthBody.Status)
+		assert.Equal(t, `{"errors":{"test":"not alive"}}`, strings.TrimSpace(string(out)))
+
+		alive = nil
+		response, err = c.Get(ts.URL + ReadyCheckPath)
+		require.NoError(t, err)
+		require.EqualValues(t, http.StatusOK, response.StatusCode)
+		require.NoError(t, json.NewDecoder(response.Body).Decode(&versionBody))
+		require.EqualValues(t, versionBody.Version, handler.VersionString)
```

Comment:
maybe we can abstract this into a helper function that is reused in the test below since they essentially should have the same assertions on basic functionality except for the middleware adding a header.
Let's keep the code DRY ;)

Reply:
done, let me know if you need something changed in that